### PR TITLE
feat: contract calls with dry-run

### LINKED ISSUE #267: Aex9 balances internal server error caused by "Out of gas"
This request returns 500:
https://mainnet.aeternity.io/mdw/aex9/balances/ct_2M4mVQCDVxu6mvUrEue1xMafLsoA1bgsfC3uT95F3r1xysaCvE

Locally it shows the reason of "Out of gas":
http://localhost:4000/aex9/balances/ct_2M4mVQCDVxu6mvUrEue1xMafLsoA1bgsfC3uT95F3r1xysaCvE

ct_2M4mVQCDVxu6mvUrEue1xMafLsoA1bgsfC3uT95F3r1xysaCvE
ct_21oSi6uMMcnnyxc1KTP8MReBPVvBQQ9oeoA7aexQBsF2o16HwT
ct_2czjMpbeeds9Geb272UfiEvhQHc8QZM5AM8jngwpqdMnWNDP3h
ct_25JPViJPDPjroiL1wr1ngWJ8pK5ykN7x3LqGMWPCuC6PtLjzKT
ct_2mzKUzT55wz4wYUmH4s3dHf3YBprDLi92uxvEah8fe3BaLfkBU
ct_2bFV4kxtmUeKTF5Eb5STfnAx6UgxzBqTL1pYLF1GBhcmoQAhLf
ct_BmCPDh6TK3KtWdwKEJrSCnrvghbkjekuTTh7B5tLGduF6rTFG
ct_227nuD7YiHL7pJ63FP1CnbhKgU9ueWKP77sA26huE47s31tuqQ
ct_jyw2fRpr6srNwSqE6FJFyZSD1upD91eZ5ZZKZfBhuDmBxTU53
ct_VxFemyM2ML15ApWrFqBKrarVbNfBASPBs1SC1zMuwaWrnrTSy
ct_2GPhKGbRxZaaJtqRTnxLk5UDSNNEtRt3tyWuKVQeoVrerRfhHa
ct_7UfopTwsRuLGFEcsScbYgQ6YnySXuyMxQWhw6fjycnzS5Nyzq
ct_M9yohHgcLjhpp1Z8SaA1UTmRMQzR4FWjJHajGga8KBoZTEPwC
ct_BwJcRRa7jTAvkpzc2D16tJzHMGCJurtJMUBtyyfGi2QjPuMVv
ct_uGk1rkSdccPKXLzS259vdrJGTWAY9sfgVYspv6QYomxvWZWBM
ct_26GAggwrHJLw45vopxu1HA51t6pw4xjzqvq66kh1WPtZjNXMyL

### PR DESCRIPTION
**Pull request by jyeshe**
## What

Replaces contract call dispatch with dry-run.

## Why

Fixes #267 

## Validation steps

`elixir --sname aeternity@localhost -S mix test.integration test/integration/ae_mdw_web/controllers/aex9_controller_test.exs`

### CODE REVIEW

File: test/integration/ae_mdw_web/controllers/aex9_controller_test.exs [hunk 1]
```diff
@@ -40,4 +42,34 @@ defmodule Integration.AeMdwWeb.Aex9ControllerTest do
...
+  end
+
+  describe "balance" do
+    test "gets an account balances for multiple contracts", %{conn: conn} do
+      account_id = "ak_WzcSck8B9ZPgHsy5XeqBbtUV4YbTuGyyJUzhSMvSK2JY1nzqJ"
+      conn = get(conn, "/aex9/balances/account/#{account_id}")
+
+      balances_response = json_response(conn, 200)
+
+      assert Enum.find(balances_response, fn %{"contract_id" => contract_id} ->
```
Comment (sborrazas) [inline test/integration/ae_mdw_web/controllers/aex9_controller_test.exs:70]: Perhaps we can use `Enum.any?/2` here instead?

File: lib/ae_mdw/dry_run/contract.ex [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+      gas: @gas,
+      gas_price: div(min_gas_price(), 1000),
+      call_data: call_data,
+      fee: 200 * min_gas_price()
+    }
+    |> :aect_call_tx.new()
+    |> Util.ok!()
+  end
+
+  def min_gas_price do
```
Comment (sborrazas) [inline lib/ae_mdw/dry_run/contract.ex:42]: Shouldn't this be private?

File: lib/ae_mdw/dry_run/contract.ex [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+    protocol = :aec_hard_forks.protocol_effective_at_height(1)
+
+    :erlang.max(
```
Comment (sborrazas) [inline lib/ae_mdw/dry_run/contract.ex:45]: You can just do `max(...)` here

File: lib/ae_mdw/dry_run/runner.ex [hunk 4]
```diff
@@ -0,0 +1,26 @@
+defmodule AeMdw.DryRun.Runner do
+  @moduledoc """
+  Simulates transactions with dry run.
+  """
```
Comment (sborrazas) [inline lib/ae_mdw/dry_run/runner.ex:4]: Not sure I fully understand what this mod does, perhaps you can explain it better? And why do we need to hard-code a pk also?

**Review by sborrazas — CHANGES_REQUESTED** [review]

File: lib/ae_mdw/dry_run/runner.ex (hunk 4 above)
Comment (jyeshe) [reply lib/ae_mdw/dry_run/runner.ex:4]: It's the concept of dry-run but after your question added info to be more clear. Better to understand?

**Review by sborrazas — APPROVED** [review]

**Pull request merged**